### PR TITLE
Improve travis-ci builds

### PR DESCRIPTION
**Pull request by wardpeet**
This has been a journey. Hopefully it's faster than before XD.

travis.yml is split in 3 stages
- prepare cache
   we build our project and upload the artifacts to gcloud
- tests
   We run all our tests in 3 steps (UNIT, SMOKE & LINT). We download all builds from gcloud and run 3 parallel tasks.
- cleanup
   We remove the glcoud artifacts

We only build PR's on node 6, master will still be build node 6,7,8

If we could cache gcloud installation as well this would help us speed up the build a bit more as well.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-matrix:
-  include:
-    - node_js: "6.9.1"
-    - node_js: "7"
-    - node_js: "8"
 dist: trusty
 env:
   global:
     - ARTIFACTS_AWS_REGION=us-west-2
     - ARTIFACTS_S3_BUCKET=lighthouse-screenshots-debugging
```
Comment (patrickhulce) [inline .travis.yml]: we can totally nix that screenshot saving too btw if we want some extra wins :)

File: .travis.yml [hunk 2]
```diff
@@ -20,34 +18,119 @@ cache:
...
-  - yarn bundlesize
-  - yarn test-cli-formatting
-  - yarn test-launcher-formatting
-  - yarn lint
-  - yarn unit
-  - yarn closure
-  - yarn smoke
-  - yarn smokehouse
-  - yarn compile-devtools
+script: "echo \"do nothing\""
```
Comment (patrickhulce) [inline .travis.yml]: maybe brief comment about the real scripts being decided per job?

File: .travis.yml [hunk 3]
```diff
@@ -20,34 +18,119 @@ cache:
...
+jobs:
+  include:
+    - &prep-cache
```
Comment (patrickhulce) [inline .travis.yml]: I had no idea this was a [thing](http://www.yaml.org/spec/1.2/spec.html#id2760395) that's amazing 😱 🎉

File: package.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
   },
   "scripts": {
     "install-all": "npm-run-posix-or-windows install-all:task",
-    "install-all:task": "yarn & yarn install-launcher & yarn install-cli & yarn install-extension & yarn install-viewer & wait",
+    "install-all:task": "yarn; yarn install-launcher & yarn install-cli & yarn install-extension & yarn install-viewer & wait",
```
Comment (patrickhulce) [inline package.json]: sidenote while you're in here: can we move `yarn;` to happen in install-all while we're at it so it doesn't fail with "npm-run-posix-or-windows not installed" :)

Comment (wardpeet) [reply package.json]: good point :)

File: .travis.yml (hunk 3 above)
Comment (wardpeet) [reply .travis.yml]: @see https://docs.travis-ci.com/user/build-stages/using-yaml-aliases/

File: .travis.yml (hunk 2 above)
Comment (wardpeet) [reply .travis.yml]: good point :)

File: .travis.yml (hunk 1 above)
Comment (wardpeet) [reply .travis.yml]: I only install it when we want to send it over so not a big issue.

File: package.json (hunk 4 above)
Comment (paulirish) [reply package.json]: yah that just means we gotta do those two sequentially...

     "install-all": "yarn; npm-run-posix-or-windows install-all:task",

**Comment by wardpeet** [comment]
Travis got updated to only run node 6 on PRs I believe this is the best thing we can improve so far so I'm closing this one.

**Pull request closed without merging**